conda/recipes/release-versions.yaml

```diff
@@ -24,7 +24,7 @@ build_stack_version:
 
 # Shared versions across meta-pkgs
 arrow_version:
-  - '=0.15.0'
+  - '=0.17.1'
```

Comment:
Note that RAPIDS 0.14 still requires arrow 0.15.0. @dillon-cullinan should this be updated to account for RAPIDS 0.15 release or stay in accordance with RAPIDS 0.14?

Reply:
@kkraus14 we have a `branch-0.14` that we use for release 0.14 so update both here as this will serve as release 0.15